src/README.md

```diff
@@ -4,76 +4,103 @@ The purpose of this document is to describe the internal code structure and majo
+- <a href="https://godoc.org/github.com/daos-stack/daos/src/control/client">DAOS management Go package</a>
+- <a href="control/dmg/README.md">DAOS Management tool (aka dmg)</a>
+
+The DAOS library (i.e. libdaos) implements the DAOS storage model and is primarily targeted at application and I/O middleware developers who want to store datasets into DAOS containers. It can be used from any nodes connected to the fabric used by the targeted DAOS system. The application process is authenticated via the DAOS agent (see next section). The API exported by libdaos is commonly called the DAOS API (in opposition to the DAOS management API) and allows to manage containers and access DAOS objects through different interfaces (e.g. key-value store or array API). The libdfs library emulates POSIX file and directory abstractions over libdaos and provides a smooth migration path for applications that requires a POSIX namespace. For further reading on libdaos, bindings for different programming languages and libdfs:
+- <a href="client/api/README.md">DAOS Library (libdaos)</a>
+- <a href="utils/py/README.md">Python API bindings</a>
+- <a href="https://github.com/daos-stack/go-daos">Go bindings</a> and <a href="https://godoc.org/github.com/daos-stack/go-daos/pkg/daos">API documentation</a>
+- <a href="client/dfs/README.md">POSIX File & Directory Emulation (libdfs)</a>
+
+The libdaos and libdfs libraries provide the foundation to support domain-specific data formats like HDF5 and Apache Arrow. For further reading on I/O middleware integration, please check the following external references:
+- <a href="https://bitbucket.hdfgroup.org/projects/HDFFV/repos/hdf5/browse?at=refs%2Fheads%2Fhdf5_daosm">DAOS VOL connector for HDF5</a>
+- <a href="https://github.com/daos-stack/mpich/tree/daos_adio">ROMIO DAOS ADIO driver for MPI-IO</a>
+
+<a id="13"></a>
+### Agent
 
-<a href="control/dmg/README.md">DAOS Management tool (aka dmg)</a>
+The <a href="control/agent/README.md">DAOS agent</a> is a daemon residing on the client node and interacts with the DAOS client library through dRPC to authenticate the application process. It is a trusted entity can sign the DAOS Client credentials using local certificates. The agent can support different authentication frameworks and uses an Unix Domain Socket by default. The DAOS agent is written in Go and communicates through gRPC with the control plane component of each DAOS server to provide DAOS system membership information to the client library and to support pool listing.
```

Comment:
"It is a trusted entity [that] can sign"

Reply:
I would change "uses an Unix Domain Socket by default" to "uses a Unix Domain Socket to communicate with the client library"